demos/coreMQTT/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1906 @@
+    /* Publish task doesn't receive anything in this demo, so it doesn't need a large queue. */
+    xPublisherResponseQueue = xQueueCreate( 1, sizeof( PublishElement_t ) );
+
+    /* In this demo, send publishes on non-subscribed topics to this queue.
+     * Note that this value is not meant to be changed after `prvCommandLoop` has
+     * been called, since access to this variable is not protected by thread
+     * synchronization primitives. */
+    xDefaultResponseQueue = xPublisherResponseQueue;
+
+    /* Connect to the broker. We connect here with the "clean session" flag set
+     * to true in order to clear any prior state in the broker. We will disconnect
+     * and later form a persistent session, so that it may be resumed if the
+     * network suddenly disconnects. */
+    LogInfo( ( "Creating a TCP connection to %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+    xNetworkStatus = prvConnectNetwork( &xNetworkContext );
+
+    ret = EXIT_FAILURE;
+
+    if( xNetworkStatus == pdPASS )
+    {
+        LogInfo( ( "Clearing broker state." ) );
+        xMQTTStatus = prvMQTTConnect( &globalMqttContext, &xNetworkContext, true );
+
+        if( xMQTTStatus == MQTTSuccess )
+        {
+            /* Disconnect. */
+            xMQTTStatus = MQTT_Disconnect( &globalMqttContext );
+
+            if( xMQTTStatus == MQTTSuccess )
+            {
+                LogInfo( ( "Disconnecting TCP connection." ) );
+                xNetworkStatus = prvDisconnectNetwork( &xNetworkContext );
+                ret = ( xNetworkStatus == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE;
+            }
+        }
+    }
+
+    if( ret == EXIT_SUCCESS )
+    {
+        /* Clear the lists of subscriptions and pending acknowledgments. */
+        memset( pxPendingAcks, 0x00, mqttexamplePENDING_ACKS_MAX_SIZE * sizeof( AckInfo_t ) );
+        memset( pxSubscriptions, 0x00, mqttexampleSUBSCRIPTIONS_MAX_COUNT * sizeof( SubscriptionElement_t ) );
+
+        /* Create inital process loop command. */
+        prvCreateCommand( PROCESSLOOP, NULL, NULL, &xCommand );
+        xResult = prvAddCommandToQueue( &xCommand );
+        ret = ( xResult == pdTRUE ) ? EXIT_SUCCESS : EXIT_FAILURE;
+    }
+
+    if( ret == EXIT_SUCCESS )
+    {
+        LogInfo( ( "Creating a TCP connection to %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+        /* Connect to the broker. */
+        xNetworkStatus = prvConnectNetwork( &xNetworkContext );
+        ret = ( xNetworkStatus == pdPASS ) ? EXIT_SUCCESS : EXIT_FAILURE;
+    }
+
+    if( ret == EXIT_SUCCESS )
+    {
+        /* Form an MQTT connection with a persistent session. */
+        xMQTTStatus = prvMQTTConnect( &globalMqttContext, &xNetworkContext, false );
+        ret = ( xMQTTStatus == MQTTSuccess ) ? EXIT_SUCCESS : EXIT_FAILURE;
+    }
+
+    if( ret == EXIT_SUCCESS )
+    {
+        configASSERT( globalMqttContext.connectStatus = MQTTConnected );
```

Comment:
```suggestion
configASSERT( globalMqttContext.connectStatus == MQTTConnected );
```
This might be missing in the FreeRTOS/FreeRTOS version?

Reply:
Yeah, it's missing there too